src/util/InputRangeUtils.h

```diff
@@ -225,4 +240,30 @@ template <typename Range, typename F>
 CachingContinuableTransformInputRange(Range&&, F)
     -> CachingContinuableTransformInputRange<all_t<Range>, F>;
 
+CPP_class_template(typename ViewType)(
+    requires(ql::ranges::input_range<ViewType>)) struct ConcatenatedInputRange
+    : ad_utility::InputRangeFromGet<ql::ranges::range_value_t<ViewType>> {
+  using ViewCollection = std::vector<std::unique_ptr<ViewType>>;
+  ViewCollection viewCollection_;
+  ql::ranges::iterator_t<ViewCollection> collectionIt_;
+
+  ConcatenatedInputRange(ViewCollection viewCollection)
+      : viewCollection_(std::move(viewCollection)),
+        collectionIt_(viewCollection_.begin()) {}
+
+  std::optional<ql::ranges::range_value_t<ViewType>> get() {
+    while (collectionIt_ != viewCollection_.end()) {
+      auto elementOpt = (*collectionIt_)->get();
+      if (elementOpt.has_value()) {
+        // Element exists in the currently iterated view
+        return elementOpt;
+      }
+      // Done with current view, but not all views
+      collectionIt_++;
```

Comment:
```suggestion
      ++collectionIt_;
```
To be preferred as sligthly more efficient in some cases.

Reply:
Done: https://github.com/ad-freiburg/qlever/pull/1960/commits/94afe8430bcdc392c190158a05d875513ccb9cab?show-viewed-files=true&file-filters%5B%5D=